src/libs/actions/IOU.js

```diff
@@ -1564,13 +1568,18 @@ function getPayMoneyRequestParams(chatReport, iouReport, recipient, paymentMetho
                 },
             },
         },
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
-            value: {
-                [optimisticReportPreviewAction.reportActionID]: optimisticReportPreviewAction,
-            },
-        },
+        ...(optimisticReportPreviewAction
+            ? [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+                    value: {
+                        [optimisticReportPreviewAction.reportActionID]: optimisticReportPreviewAction,
+                    },
+                },
+              ]
+            : []
+        ),
```

Comment:
I'm not really a fan of adding the spread operator, and the ternary, and the empty arrays. I think that makes it very confusing to know what's going on. What I would suggest is to do something more like:

```
if (optimisticReportPreviewAction) {
    optimisticData.push({
        onyxMethod: Onyx.METHOD.MERGE,
        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
        value: {
            [optimisticReportPreviewAction.reportActionID]: optimisticReportPreviewAction,
        },
    });
}
```